docs/source/python-server-api-examples.md

```diff
@@ -750,9 +749,13 @@ threshold_tx_transfer
 
 ### Hash-locked Conditions
 
-By creating a hash of a difficult-to-guess 256-bit random or pseudo-random integer it is possible to create a condition which the creator can trivially fulfill by publishing the random value. However, for anyone else, the condition is cryptographically hard to fulfill, because they would have to find a preimage for the given condition hash.
+A hash-lock condition on an asset is like a password condition: anyone with the secret preimage (like a password) can fulfill the hash-lock condition and transfer the asset to themselves.
+
+Under the hood, fulfilling a hash-lock condition amounts to finding a number (a "preimage") which, when hashed, results in a given value. It's easy to verify that a given preimage hashes to the given value, but it's computationally difficult to _find_ a number which hashes to the given value. The only practical way to get a valid preimage is to get it from the original creator (possibly via intermediaries).
```

Comment:
I suspected that any arbitrary string could be the preimage, but the original text used the words "integer" and "value". I will push a change to this.